domain/src/main/kotlin/no/nav/su/se/bakover/domain/oppdrag/Utbetalingsstrategi.kt

```diff
@@ -129,15 +133,56 @@ sealed class Utbetalingsstrategi {
             )
         }
 
-        private fun createUtbetalingsperioder(beregning: Beregning) =
-            SlåSammenEkvivalenteMånedsberegningerTilBeregningsperioder(beregning.getMånedsberegninger()).beregningsperioder
-                .map {
+        private fun createUtbetalingsperioder(
+            beregning: Beregning,
+            uføregrunnlag: List<Grunnlag.Uføregrunnlag>,
+        ): List<Utbetalingsperiode> {
+            val slåttSammenMånedsberegninger = SlåSammenEkvivalenteMånedsberegningerTilBeregningsperioder(
+                beregning.getMånedsberegninger(),
+            ).beregningsperioder
+
+            val slåttSammenUføregrunnlag =
+                uføregrunnlag.groupByContinuous()
+
+            val månedsperioder = slåttSammenMånedsberegninger.map {
+                it.periode
+            }.flatMap {
+                it.tilMånedsperioder()
+            }.distinct()
+
+            val uføregrunnlagPerioder = slåttSammenUføregrunnlag.map {
+                it.first
+            }.flatMap {
+                it.tilMånedsperioder()
+            }.distinct()
+
+            val uføregrunnlagInneholderAlleMånedsperioder = månedsperioder.all {
+                uføregrunnlagPerioder.contains(it)
+            }
+
+            val månedsperioderInneholderAlleUføreperioder = uføregrunnlagPerioder.all {
+                månedsperioder.contains(it)
+            }
+
+            if (!uføregrunnlagInneholderAlleMånedsperioder || !månedsperioderInneholderAlleUføreperioder) {
+                throw UtbetalingStrategyException("Fant ikke uføregrad for en månedsperiode")
+            }
+
+            return slåttSammenUføregrunnlag.flatMap { grunnlag ->
+                slåttSammenMånedsberegninger.mapNotNull { månedsbereging ->
+                    månedsbereging.periode.snitt(grunnlag.first)?.let {
+                        månedsbereging to grunnlag.second
```

Comment:
Har vi god nok testdekning på denne? Burde det ikke vært `it to grunnlag.second`? Og i så fall, lag en test først

Reply:
hvis vi bruker it her, så får vi ikke månedsberegninger i mappen under for å få tak i beløpet.